src/uct/gaudi/base/gaudi_topo.c

```diff
@@ -0,0 +1,1919 @@
+        ucs_free(path);
+        return UCS_ERR_INVALID_PARAM;
+    }
+    ucs_free(path);
+
+    *vendor_id = (uint16_t)val;
+    return UCS_OK;
+}
+
+/* Helper function to read PCI address from sysfs */
+static ucs_status_t
+uct_gaudi_read_pci_addr(const char *accel_name, char *pci_addr, size_t max)
+{
+    char path[PATH_MAX];
+    FILE *file;
+    ucs_status_t status;
+    size_t len;
+
+    ucs_snprintf_safe(path, PATH_MAX, "%s%s/device/pci_addr",
+                      UCT_GAUDI_TOPO_ACCEL_PATH, accel_name);
+
+    file = fopen(path, "r");
+    if (!file) {
+        ucs_debug("Failed to open %s", path);
+        return UCS_ERR_IO_ERROR;
+    }
+
+    if (fgets(pci_addr, max, file) == NULL) {
+        ucs_debug("Failed to read PCI address from %s", path);
+        status = UCS_ERR_IO_ERROR;
+    } else {
+        /* Remove trailing newline */
+        len = strlen(pci_addr);
+        if (len > 0 && pci_addr[len - 1] == '\n') {
+            pci_addr[len - 1] = '\0';
+        }
+        status = UCS_OK;
+    }
+
+    fclose(file);
+    return status;
+}
+
+/* Helper function to read module ID from sysfs */
+static ucs_status_t
+uct_gaudi_read_module_id(const char *accel_name, uint32_t *module_id)
+{
+    char path[PATH_MAX];
+    FILE *file;
+    char buffer[16];
+    char *endptr;
+    unsigned long val;
+
+    ucs_snprintf_safe(path, PATH_MAX, "%s%s/device/module_id",
+                      UCT_GAUDI_TOPO_ACCEL_PATH, accel_name);
+
+    file = fopen(path, "r");
+    if (!file) {
+        ucs_debug("Failed to open %s", path);
+        return UCS_ERR_IO_ERROR;
+    }
+
+    if (fgets(buffer, sizeof(buffer), file) == NULL) {
+        ucs_debug("Failed to read module ID from %s", path);
+        fclose(file);
+        return UCS_ERR_IO_ERROR;
+    }
+    fclose(file);
+
+    errno = 0;
+    val   = strtoul(buffer, &endptr, 10);
+    while (isspace((unsigned char)*endptr)) {
+        endptr++;
+    }
+    if (errno != 0 || endptr == buffer || *endptr != '\0') {
+        ucs_debug("Invalid module ID in %s: '%s'", path, buffer);
+        return UCS_ERR_INVALID_PARAM;
+    }
+    if (val > UINT32_MAX) {
+        ucs_debug("Module ID %lu exceeds uint32_t in %s", val, path);
+        return UCS_ERR_INVALID_PARAM;
+    }
+
+    *module_id = (uint32_t)val;
+    return UCS_OK;
+}
+
+/* Get Gaudi device index from a given module ID. */
+int uct_gaudi_get_index_from_module_id(uint32_t module_id)
+{
+    DIR *dir;
+    struct dirent *entry;
+    ucs_status_t status;
+    uint32_t read_module_id;
+    int device_id;
+
+    dir = opendir(UCT_GAUDI_TOPO_ACCEL_PATH);
+    if (!dir) {
+        ucs_error("Failed to open directory %s: %s", UCT_GAUDI_TOPO_ACCEL_PATH,
+                  strerror(errno));
+        return -1;
+    }
+
+    /* Default: not found */
+    device_id = -1;
+    while ((entry = readdir(dir)) != NULL) {
+        if (strncmp(entry->d_name, "accel", 5) != 0 ||
+            strncmp(entry->d_name, "accel_", 6) == 0) {
```

Comment:
use ( ) around if conditions
```suggestion
        if ((strncmp(entry->d_name, "accel", 5) != 0) ||
            (strncmp(entry->d_name, "accel_", 6) == 0)) {
```